\subsection{Introduction}

Ligand binding sites are protein regions that interact with other biochemical entities such as peptides or organic small molecules. The binding process eventually results in a selective modulation of the protein function. Indeed, one of the most successful strategies in conventional drug discovery is to identify, based on the high-resolution three-dimensional structure of binding sites, small molecules that activate or inhibit a protein associated with a disease\cite{sadybekov_computational_2023}.

Alongside the increasing number of available protein structures in the Protein Data Bank (PDB)\cite{goodsell_rcsb_2020}, structure-based approaches have become a crucial computational framework in early stages of drug development\cite{batool_structure-based_2019, sledz_protein_2018}. By focusing on ligand binding sites, such strategies enable a rational design and optimization of drugs and reduce the probability of failure of those compounds that reach clinical trials\cite{macalino_role_2015}. Protein-small molecule docking is among the most popular structure-based strategies to predict drug-target interactions, and it aims at finding the optimal location and conformation of a given ligand with respect to the receptor binding site\cite{kitchen_docking_2004}. Molecular docking has been successfully applied in proteome-scale studies (e.g. reverse screening\cite{westermaier_virtual_2015, lee_using_2016, pinzi_molecular_2019}), but the target-dependent nature of scoring functions prevents the direct comparisons of docking results across different proteins and protein families. Indeed, the design of a universal docking scoring function still remains a challenge\cite{li_overview_2019, shen_machine_2020}. Consequently, alternative strategies such as reverse pharmacophore screening, binding site similarity assessment or interaction fingerprint comparison are often employed in proteome-wide analyses\cite{sydow_advances_2019}.

Most of these approaches require the detailed characterization of protein binding sites in a machine-readable format suitable for computational applications, which reasonably allows for the possibility of borrowing featurization techniques from related fields. In fact, characterizing small molecules through numerical vectors encoding topological or physicochemical properties is a very common strategy in cheminformatics, and sets the stage for many drug discovery projects founded on the small-molecule similarity principle\cite{fernandez-torras_connecting_2022, cereto-massague_molecular_2015, muegge_overview_2016}. Likewise, descriptors for larger molecules, such as protein targets, can also be derived, usually gathering features from their amino acid sequences\cite{bileschi_using_2022}. However, the exploitation of structural data offers a complementary perspective to create protein descriptors and is therefore more promising than the treatment of protein sequences alone. Indeed, biophysical interactions between proteins and ligands occur in very specific areas of protein surfaces (i.e. binding sites) and involve a limited set of residues, which has driven the development of structure-based protein descriptors focused on these particular regions\cite{eguida_estimating_2022}.

Pocket descriptors are commonly classified according to the underlying binding site representation they consider, often based on binding site residues (e.g. FuzCav\cite{weill_alignment-free_2010}, SiteAlign\cite{schalon_simple_2008}), pocket surfaces (e.g. MaSIF\cite{gainza_deciphering_2020}) or explicit interactions with bound ligands or probes (e.g. KRIPO\cite{wood_pharmacophore_2012}, TIFP\cite{desaphy_encoding_2013}, BioGPS\cite{siragusa_biogps_2015}). In addition, and together with the rising interest in deep learning applications in drug discovery\cite{chen_rise_2018}, novel data-driven approaches have been designed to derive pocket descriptors borrowing techniques from computer vision (e.g. DeeplyTough\cite{simonovsky_deeplytough_2020}, BindSiteS-CNN\cite{scott_classification_2022}). 

Apart from the inherent characterization of binding sites, pocket descriptors provide an excellent means to estimate binding site similarity, which is thus simplified into straightforward vector distance measurements. Binding site comparisons (aka pocket matching) have emerged as a promising methodology to move away from the ‘one drug-one target-one disease’ paradigm\cite{morphy_magic_2004} by assessing complex studies involving multiple-target drug binding events\cite{konc_binding_2019, naderi_binding_2019, zhang_computational_2017}. Binding site similarity is reported to play an important role in the evaluation of ligand promiscuity\cite{haupt_drug_2013} and in the prediction of protein function, enabling the identification of similar binding sites in proteins having no sequence nor fold similarity\cite{konc_binding_2014}. Indeed, the detection of similar binding sites was helpful in several drug repurposing and polypharmacology studies\cite{duran-frigola_detecting_2017, ehrt_impact_2016, jalencas_identification_2013, salentin_polypharmacology_2014, zhao_delineation_2016} and in the prediction of possible distant drug off-targets\cite{schumann_identification_2013}. In addition, encoding pockets as numerical descriptors entails the possibility of integrating them in a unified framework together with a rich portrait of biochemical entities described in a common vectorial format, such as small molecules, cell lines or diseases\cite{fernandez-torras_connecting_2022}. For instance, chemogenomic studies are often addressed by the combination of protein descriptors and molecular fingerprints, usually referred to as proteochemometric (PCM) approaches\cite{bongers_proteochemometrics_2019, dsouza_machine_2020}.

However, existing methods to generate pocket descriptors exhibit several intrinsic limitations. One of their main drawbacks is the need of co-crystallized ligands to effectively recognize the most relevant biophysical interactions occurring in the binding site, which restricts the applicability domain of such methods to holo structures\cite{wood_pharmacophore_2012, desaphy_encoding_2013}. Another important issue related with pocket descriptors is the handcrafted nature of considered binding site representations, often selecting parameters based on specific datasets and performing poorly when used in more general and diverse scenarios\cite{ehrt_benchmark_2018}. Moreover, several strategies also rely on alignment-dependent comparisons, which makes them particularly useful to provide significant insights into the underlying patterns rationalizing binding site similarity, but also come together with an increased computational cost\cite{schalon_simple_2008}. In addition, those approaches built upon deep learning algorithms also suffer from lack of interpretability, a well-known problem in the field\cite{jimenez-luna_artificial_2021, vamathevan_applications_2019, ching_opportunities_2018}. Finally, the availability of three-dimensional protein structures has traditionally been the main limiting factor in structure-based drug discovery, but this is no longer the case. In the era of accurate protein structure prediction\cite{jumper_highly_2021, tunyasuvunakool_highly_2021, varadi_alphafold_2022}, where exhaustive collections of predicted structures are available for both relevant organisms\cite{david_alphafold_2022} and sequences derived from metagenomic studies\cite{lin_evolutionary-scale_2022}, the structural characterization of proteins is now feasible for essentially any protein sequence of interest. Accordingly, the use of pocket descriptors opens the possibility of characterizing complete proteomes and charting the pocket space in a similar way molecular fingerprints enable the exploration of the chemical space of small molecules\cite{lipinski_navigating_2004, willett_similarity-based_2006, capecchi_one_2020}.

To partially overcome the aforementioned limitations of existing pocket descriptors, we exploit the assumption that similar pockets bind similar ligands, which should result in similar rankings in a structure-based virtual screening of small molecules. Indeed, Govindaraj and Brylinski\cite{govindaraj_comparative_2018} showed that docking scores tended to be more correlated in pockets binding to chemically similar ligands than in pockets binding to dissimilar ligands. This opens the possibility of estimating binding site similarity on the basis of docking rankings and enrichments, as explored by Schmidt and co-workers in their analysis of the human kinome\cite{schmidt_analyzing_2021}. Moreover, inverse virtual screening (i.e. the screening of a set of targets for a query ligand) has been recently applied to distinguish nucleotide and heme-binding sites from a control set of pockets\cite{pu_deepdrug3d_2019}. In view of these results, we hypothesized that virtual screening could represent a promising strategy to generate pocket descriptors.

Here, we present PocketVec, a novel strategy to generate interpretable and fixed-length protein binding site descriptors based on the assumption that similar pockets bind similar ligands. Our approach is built upon inverse virtual screening, i.e. the prioritization of a given set of small molecules is expected to be more correlated between similar pockets than between dissimilar ones. We implement and assess the accuracy of our method and the derived pocket descriptors on several predefined benchmark sets. Additionally, we use bound ligands and pocket detection algorithms to comprehensively identify drug binding pockets in experimentally determined and AF2 predicted structures in the human proteome, and derive PocketVec descriptors for all identified pockets. We finally use PocketVec descriptors to exhaustively compare all pockets found in experimental and AF2 structures, explore potential relationships between pocket similarity and small molecule binding and assess a possible complementarity with other sequence and structure-based approaches to demonstrate its potential to find and characterize similar binding sites in unrelated proteins.