\subsection{Introduction}

Small molecule descriptors are crucial in cheminformatics-related tasks and provide a computational basis to guide medicinal chemistry efforts\cite{mcgibbon_intuition_2023}. Indeed, chemical libraries are growing exponentially in the last decades and are eager for efficient approaches to characterize and explore them \cite{kim_pubchem_2023, irwin_zinc20free_2020}.

Typically, small molecule descriptors are high-dimensional numerical vectors, which makes their visual representation impractical for human interpretation \cite{cereto-massague_molecular_2015}. However, several dimensionality reduction techniques, such as PCA, can be applied to compress signatures into lower-dimensional spaces (usually in two dimensions), allowing for the abstract yet informative visualization of compound collections in a directly interpretable figure \cite{probst_visualization_2020, akella_cheminformatics_2010}. The t-distributed stochastic neighbor embedding (t-SNE) technique \cite{van_der_maaten_laurens_geoffrey_hinton_visualizing_2008}, particularly suited to preserve the local structure of the data (i.e. similar compounds are placed closely in the compressed space), is one of the most common approaches to comprehensively depict collections of small molecules. 

While some chemical libraries have a narrow scope and include a limited number of compounds (e.g. covalent inhibitors \cite{du_covalentindb_2021}), many others aim to maximize the coverage of the chemical space, usually encompassing large and ever-expanding collections of small molecules \cite{hoffmann_next_2019}. In this context, it is often necessary to reach a fair balance between library size and scope. The most straightforward library filters are usually based on physicochemical properties or drug-likeness criteria, in order to maximize the eventual probability of clinical success. However, to minimize the redundancy within a set of small molecules (i.e. reducing the number of similar compounds while preserving the overall chemical diversity), clustering methods offer a more refined approach compared to basic filtering techniques \cite{lipkowitz_clustering_2002}. Such methods are typically coupled with compound chemical descriptors and are designed to select a representative set of compounds from the original set. 

With the growing number of small molecules available in public and proprietary libraries, a significant amount of biological data has been generated and associated with many of these compounds \cite{zdrazil_chembl_2024, kim_pubchem_2023}. In fact, such data has revealed relationships between compounds that extend beyond their chemical features, driving the field towards a more integrative and holistic view. For instance, molecules exhibiting similar sensitivity profiles or drugs showing similar side effects tended to share mechanisms of action, even if they were structurally dissimilar\cite{campillos_drug_2008}.

Based on these observations, our lab recently introduced the Chemical Checker (CC), a resource that integrates major chemogenomics and drug activity repositories to provide the largest collection of small molecule bioactivity signatures available to date. Beyond chemical structures and properties, CC bioactivity signatures also characterize compounds based on their target binding profiles, induced gene expression changes, and clinical annotations, among many others. For a more detailed and comprehensive description of the Chemical Checker database and signatures, we refer the reader to the \hyperref[Introduction_extending]{Introduction section} (1.7 Extending the similarity principle beyond chemical structures) and \hyperref[Chapter_3.1]{Chapter 3.1}. 

However, such bioactivity descriptors remain scarce, mainly limited to a definite set of well-characterized compounds. To overcome this limitation, our lab developed and trained a collection of deep neural networks able to leverage the experimentally determined bioactivity data associated with small molecules to infer the missing bioactivity signatures for any compound of interest\cite{bertoni_bioactivity_2021}. Our strategy related to 25 bioactivity types, as described in the original Chemical Checker publication \cite{duran-frigola_extending_2020}.

The work presented here illustrates several projects and exercises common in the field of cheminformatics, involving both chemical and bioactivity signatures. Although the presented approaches and results are independent of each other, the use of small molecule signatures constitutes a central theme, a shared foundation in which all projects are anchored. First, we signaturize and cluster a chemical library of small molecules to reduce its size while preserving the overall chemical diversity of the original set. We then illustrate how different chemical libraries may occupy distinct areas of the chemical space, highlighting the dissimilarities between natural and synthetic compounds. Finally, we show how bioactivity signatures may uncover relationships between molecules that can not be explained by chemical properties alone. 

Some of the presented work is a direct result of external collaborations with other research labs and private companies. A brief explanation of such collaborations is included in the \hyperref[Navigation_SupplementaryInformation]{Supplementary Information}. 




