\subsection{Introduction}

Ligand binding sites are protein regions that interact with other biochemical entities such as peptides or organic small molecules. The binding process eventually results in a selective modulation of the protein function. Indeed, one of the most successful strategies in conventional drug discovery is to identify, based on the high-resolution three-dimensional structure of binding sites, small molecules that activate or inhibit a protein associated with a disease\cite{sadybekov_computational_2023}.

Alongside the increasing number of available protein structures in the Protein Data Bank (PDB)\cite{goodsell_rcsb_2020}, structure-based approaches have become a crucial computational framework in early stages of drug development\cite{batool_structure-based_2019, sledz_protein_2018}. By focusing on ligand binding sites, such strategies enable a rational design and optimization of drugs and reduce the probability of failure of those compounds that reach clinical trials\cite{macalino_role_2015}. Protein-small molecule docking is among the most popular structure-based strategies to predict drug-target interactions, and it aims at finding the optimal location and conformation of a given ligand with respect to the receptor binding site\cite{kitchen_docking_2004}. Molecular docking has been successfully applied in proteome-scale studies (e.g. reverse screening\cite{westermaier_virtual_2015, lee_using_2016, pinzi_molecular_2019}), but the target-dependent nature of scoring functions prevents the direct comparisons of docking results across different proteins and protein families. Indeed, the design of a universal docking scoring function still remains a challenge\cite{li_overview_2019, shen_machine_2020}. Consequently, alternative strategies such as reverse pharmacophore screening, binding site similarity assessment or interaction fingerprint comparison are often employed in proteome-wide analyses\cite{sydow_advances_2019}.

Most of these approaches require the detailed characterization of protein binding sites in a machine-readable format suitable for computational applications, which reasonably allows for the possibility of borrowing featurization techniques from related fields. In fact, characterizing small molecules through numerical vectors encoding topological or physicochemical properties is a very common strategy in cheminformatics, and sets the stage for many drug discovery projects founded on the small-molecule similarity principle\cite{fernandez-torras_connecting_2022, cereto-massague_molecular_2015, muegge_overview_2016}. Likewise, descriptors for larger molecules, such as protein targets, can also be derived, usually gathering features from their amino acid sequences\cite{bileschi_using_2022}. However, the exploitation of structural data offers a complementary perspective to create protein descriptors and is therefore more promising than the treatment of protein sequences alone. Indeed, biophysical interactions between proteins and ligands occur in very specific areas of protein surfaces (i.e. binding sites) and involve a limited set of residues, which has driven the development of structure-based protein descriptors focused on these particular regions\cite{eguida_estimating_2022}.

Pocket descriptors are commonly classified according to the underlying binding site representation they consider, often based on binding site residues (e.g. FuzCav\cite{weill_alignment-free_2010}, SiteAlign\cite{schalon_simple_2008}), pocket surfaces (e.g. MaSIF\cite{gainza_deciphering_2020}) or explicit interactions with bound ligands or probes (e.g. KRIPO\cite{wood_pharmacophore_2012}, TIFP\cite{desaphy_encoding_2013}, BioGPS\cite{siragusa_biogps_2015}). In addition, and together with the rising interest in deep learning applications in drug discovery\cite{chen_rise_2018}, novel data-driven approaches have been designed to derive pocket descriptors borrowing techniques from computer vision (e.g. DeeplyTough\cite{simonovsky_deeplytough_2020}, BindSiteS-CNN\cite{scott_classification_2022}). 